crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs

```diff
@@ -82,8 +82,9 @@ where
     if ca.is_empty() {
         return Ok(Series::new_empty(ca.name(), ca.dtype()));
     }
-    let ca = ca.rechunk();
-    let by = by.rechunk();
+    if by.null_count() > 0 {
```

Comment:
This could be factored out of branch and be checked before any compute (rechunking).